Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadColorUnpacker.cs

```diff
@@ -10,21 +12,61 @@ namespace Speckle.Connectors.Autocad.HostApp;
 /// <summary>
 /// Expects to be a scoped dependency for a given operation and helps with layer creation and cleanup.
 /// </summary>
-public class AutocadColorManager
+public class AutocadColorUnpacker
 {
-  // POC: Will be addressed to move it into AutocadContext!
-  private Document Doc => Application.DocumentManager.MdiActiveDocument;
+  private readonly ILogger<AutocadColorUnpacker> _logger;
 
-  /// <summary>
-  /// For receive operations
-  /// </summary>
-  public Dictionary<string, AutocadColor> ObjectColorsIdMap { get; } = new();
+  public AutocadColorUnpacker(ILogger<AutocadColorUnpacker> logger)
+  {
+    _logger = logger;
+  }
 
   /// <summary>
   /// For send operations
   /// </summary>
   private Dictionary<string, ColorProxy> ColorProxies { get; } = new();
 
+  /// <summary>
+  /// Iterates through a given set of autocad objects and layers to collect colors.
+  /// </summary>
+  /// <param name="unpackedAutocadRootObjects">atomic root objects, including instance objects</param>
+  /// <param name="layers">layers used by atomic objects</param>
+  /// <returns></returns>
+  public List<ColorProxy> UnpackColors(
+    List<AutocadRootObject> unpackedAutocadRootObjects,
+    List<LayerTableRecord> layers
+  )
+  {
+    // Stage 1: unpack colors from objects
+    foreach (AutocadRootObject rootObj in unpackedAutocadRootObjects)
+    {
+      try
+      {
+        Entity entity = rootObj.Root;
+        ProcessObjectColor(rootObj.ApplicationId, entity.Color);
+      }
+      catch (Exception e) when (!e.IsFatal())
+      {
+        _logger.LogError(e, "Failed to unpack colors from Autocad Entity."); // TODO: Check with Jedd!
```

Comment:
Same with several of these classes. Prefer` IReadOnlyCollection` over `List`, `ex` instead of `e`, and no trailing `.` on logger templates.